packages/astro/src/core/ssr/css.ts

```diff
@@ -25,7 +25,7 @@ export function getStylesForURL(filePath: URL, viteServer: vite.ViteDevServer):
       if (!importedModule.id || scanned.has(importedModule.id)) continue;
       const ext = path.extname(importedModule.id.toLowerCase());
       if (STYLE_EXTENSIONS.has(ext)) {
-        css.add(importedModule.id); // if style file, add to list
+        css.add(importedModule.url || importedModule.id); // if style file, add to list
```

Comment:
In Vite’s module graph, `id` is usually a full-path URLs (`C:/Users/astro/code/my-project/src/components/React.css`). But `url` is usually a project-relative URL (`/src/components/React.css`). This prefers the latter, if available.

Reply:
The strange part is that the former does work in most instances! That’s why we used it in the first place. But I suppose `url`s are still generally more reliable.